### DISCUSSION TITLE
How to use an action function inside a pathless routes?

### DISCUSSION THREAD

**Discussion by rodolphoasb** [remix-run/remix#7708] [Q&A]
I have this structure:

![image](https://github.com/remix-run/remix/assets/8127858/1c87327f-3581-4669-b198-34a4d04847cb)

`_db` It's a dashboard layout that has a signout button. When I click on the signout button, I always get: 

`405 Method Not Allowed`.
Route "routes/_db.home._index" does not have an action, but you are trying to submit to it. To fix this, please add an `action` function to the route

But this form action is inside the `_db` folder. Why this doesn't work?

**Comment by rhyek** [comment]
did you get it working?

**Reply by kiliman** [reply]
Pathless layouts can't have an action. Remix only POSTs to a single route, unlike loaders. Since pathless layouts are not part of the URL, Remix doesn't know which route to dispatch to.

I would recommend just creating a `signout.tsx` route, then for your sign out button, set `<Form action="/signout" method="post">`

**Reply by 03xb7** [reply]
Thanks for that insight, I had the exact same issue as the OP and didn't know pathless layouts were disregarded when resolving actions.
